docs/contributing/03-stories/story-16.md

```diff
@@ -0,0 +1,100 @@
+# User Story 16 - Wing for TypeScript
+
+> **Status**: Draft
+
+Anders loves TypeScript, but he also loves the amazing experience Wing offers for building cloud applications.
+
+It's Andres' lucky day because Wing Cloud had just announced that very soon TypeScript developers are able to start leveraging Wing libraries from the comfort of
+their favorite programming language.
+
+He got an invite for a pre-release. The invite explains that initially, there is only support for `cloud.Bucket` and `cloud.Function` and for consuming [winglibs](https://github.com/winglang/winglibs).
+
+The example looks like this:
+
+First, install some libraries:
+
+```sh
+$ npm i winglang wingsdk @winglibs/cloud @winglibs/checks
+```
+
+```ts
+// main.ts
+import { inflight, App } from "wingsdk";
+import * as cloud from "@winglibs/cloud";
+import { Check } from "@winglibs/checks";
+
+// Final App depends on platform used
+const app = new App();
+
+const data = "world";
+const bucket = new cloud.Bucket(app, "Bucket");
+
+const fn = new cloud.Function(app, "Function", inflight({ bucket, data }, async (ctx, event) => {
+  await ctx.bucket.put("hello.txt", ctx.data);
+});
+
+bucket.grantPut(fn);
```

Comment:
implementation note: This seems fine with me and the idea of automatically codegenerating these based on available inflight methods makes sense

Reply:
yes that will be awesome.

I would implement these manually for now in `@winglibs/cloud` but I think we will need to expose some APIs from the standard library to be able to explicitly add bindings/permissions/lifting operations.

@Chriscbr see above.

Reply:
added some pseudo code in the winglib code.